Formula/xgboost.rb

```diff
@@ -13,10 +13,19 @@ class Xgboost < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on "libomp"
```

Comment:
Based on my currently incomplete understanding of how Homebrew, OpenMP and formulae all interact, I think this formula should be compiled with GCC, and pull in GCCs native `libgomp`. Here is the relative text from a discussion among maintainers that I feel pretty OK/confident is correct:

> The current stand on OpenMP is: use Apple’s clang + libomp (which is a hack) only for standalone formulas that strongly benefit from OpenMP and do not expose this to dependents (good example being imagemagick).

So if we take that as truth I think you need to do:

```suggestion
  depends_on "gcc"
  
  fails_with :clang # because this is an OpenMP enabled library like OpenBLAS
```